src/SFA.DAS.Forecasting.Domain/Commitments/EmployerCommitments.cs

```diff
@@ -4,41 +4,89 @@
+                .Where(m => m.SendingEmployerAccountId == EmployerAccountId)
+                .Where(m => m.FundingSource == FundingSource.Transfer)
+                .ToList()
+                .AsReadOnly();
         }
 
-        public virtual Tuple<decimal, List<long>> GetTotalCostOfTraining(DateTime date)
+        public virtual CostOfTraining GetTotalCostOfTraining(DateTime date)
         {
-            var startOfMonth = date.GetStartOfMonth();
-            var commitments = _commitments.Where(commitment =>
-                    commitment.StartDate.GetStartOfMonth() < startOfMonth &&
-                    commitment.PlannedEndDate.GetLastPaymentDate().GetStartOfMonth() >= startOfMonth)
-                .ToList();
-            return new Tuple<decimal, List<long>>(commitments.Sum(c => c.MonthlyInstallment), commitments.Select(c => c.Id).ToList());
+            bool FilterCurrent(CommitmentModel c) =>
+                      c.StartDate.GetStartOfMonth() < date.GetStartOfMonth() &&
+                      c.PlannedEndDate.GetLastPaymentDate().GetStartOfMonth() >= date.GetStartOfMonth();
+
+            var levyFundedCommitments = _levyFundedCommitments.Where(FilterCurrent);
+            var sendingEmployerCommitments = _sendingEmployerTransferCommitments.Where(FilterCurrent);
+            var receivingEmployerCommitments = _receivingEmployerTransferCommitments.Where(FilterCurrent);
+
+            var includedCommitments = new List<CommitmentModel>();
+            includedCommitments.AddRange(levyFundedCommitments);
+            includedCommitments.AddRange(sendingEmployerCommitments);
+            includedCommitments.AddRange(receivingEmployerCommitments);
+
+            return new CostOfTraining
+            {
+                LevyFunded = levyFundedCommitments.Sum(c => c.MonthlyInstallment),
+                TransferIn = receivingEmployerCommitments.Sum(m => m.MonthlyInstallment),
+                TransferOut = sendingEmployerCommitments.Sum(m => m.MonthlyInstallment) + receivingEmployerCommitments.Sum(c => c.MonthlyInstallment),
+                CommitmentIds = includedCommitments.Select(c => c.Id)
+            };
         }
 
-        public virtual Tuple<decimal, List<long>> GetTotalCompletionPayments(DateTime date)
+        public virtual CompletionPayments GetTotalCompletionPayments(DateTime date)
         {
-            var startOfMonth = date.GetStartOfMonth();
-            var commitments = _commitments.Where(commitment =>
-                   commitment.PlannedEndDate.GetStartOfMonth().AddMonths(1) == startOfMonth)
-                .ToList();
-            return new Tuple<decimal, List<long>>(commitments.Sum(c => c.CompletionAmount), commitments.Select(c => c.Id).ToList());
+            bool FilterCurrent(CommitmentModel c) => c.PlannedEndDate.GetStartOfMonth().AddMonths(1) == date.GetStartOfMonth();
+
+            var levyFundedCommitments = _levyFundedCommitments.Where(FilterCurrent);
```

Comment:
use .Where(...).ToList() to make it explicit that this list won't be enumerated multiple times.